### ISSUE TITLE
Self-Hosting docs + questions

### ISSUE THREAD

**Issue by Gowiem** [snaplet/docs-old#169]
## Info

My firm is working on a project for a large enterprise client and we're evaluating Snaplet. We're working to put together a POC of Snaplet against the client's real database and give it to the Dev team to see if it meets their needs. Our client has mentioned that they would like us to self-host the POC, which is understandable as their data is sensitive. 

In looking at the docs and playing around, I had a couple questions to help flesh out our understanding better:

1. When capturing a snapshot locally (via supplying a local path to the `capture` sub-command), that captures it onto the executors machine. You then go on to say "you now have a snapshot that you can upload and share with your team". Our question is: Is there any built in way to point the Snaplet at a particular repository for pulling those shared snapshots? Maybe a snaplet configuration or CLI option that will download a snapshot from a given URL? If this is not an option today, will it be supported in the future? 
2. We're getting into a continuous loop when sampling %1 of our `account` table. It just continually spins on 3 tables like so: 

```
- db.account...
- db.account...
- db.account_source...
- db.account_status...
- db.account...
- db.account...
- db.account_source...
- db.account_status...
```

I have the following `snaplet.config.ts` in the root of our repo: 

```typescript
import { defineConfig } from "snaplet";
export default defineConfig({
  sample: {
    targets: [
      {
        table: "db.account",
        percent: 1
      },
    ],
    followNullableRelations: false,
    maxChildrenPerNode: 10,
    maxCyclesLoop: 0,
    keepDisconnectedTables: true,
  },
});
```

Any additional way to debug or avoid that continuous cycling?

**Comment by peterp** [comment]
Hey @Gowiem,

> Is there any built in way to point the Snaplet at a particular repository for pulling those shared snapshots?

At the moment no, but we wrote the code for finding snapshots to be extendable, and we will likely offer more solutions for self-hosting in the future. Snaplet also offers public/ private key encryption: So it's possible to capture in your own infrastructure, encrypt the snapshots, and then share them with Snaplet Cloud so that they're discoverable by members in a team. I understand that this might not be desirable, but it's an option for the moment that'll have the least amount of friction, and a higher level of security.

> We're getting into a continuous loop when sampling %1 of our account table. 

We will 100% fix this for you as soon as possible. I'm looping in @avallete, @jgoux & @CarelFdeWaal  who wrote the majority of this code. 

If you have Discord it would be great to join us on our server, https://app.snaplet.dev/chat, so that we can help you in realtime. Otherwise. Please feel free to drop me an email at `peter@snaplet.dev`

**Comment by avallete** [comment]
Hey @Gowiem 

Since you're using the local capture we might need some additional details from you to try to debug this issue.

First, would it be possible to provide us the `schema.sql` generated so we can analyse the ERD of the relationships in the database and try to reproduce a similar one ?

Second, could you try to run the capture with the debug flag: `DEBUG=snaplet:capture* snaplet ss capture` ?
That way we could have a step by step about what the algorithm is doing and over which relationship exactly he cycles on.

Also it might be good to try to run the debug with the following target instead `{ table: "db.account", rowLimit: 1}` that way, the algorithm will start from one single row, this should avoid too much noise in the debug logs and help us identify if the issue come only from the relationship constraints or if the data in it are somehow involved as well.

**Comment by Gowiem** [comment]
@avallete happy to debug this further, but I don't think I can provide the debug output or shema due to privacy concerns for my client. Could we do a quick call to debug this and I'll be able to share details there? I'll start a thread in Discord 👍 

FYI -- I ran with the `DEBUG` output env var you mentioned. Nothing jumped out that seemed useful to share AFAICT. I also set the `rowLimit` to 1 for that single table and it ended up still pulling all `account` rows. 

Thanks for the help!
